texar/data/data/multi_aligned_data.py

```diff
@@ -0,0 +1,701 @@
+                    for sent in transposed_example
+                )
+
+                text_ids = torch.from_numpy(text_ids).to(device=self.device)
+                lengths = torch.tensor(lengths, dtype=torch.long,
+                                       device=self.device)
+
+                batch.update(
+                    {connect_name(self._name_prefix[i], "text"):
+                     transposed_example,
+                     connect_name(self._name_prefix[i], "text_ids"):
+                     text_ids,
+                     connect_name(self._name_prefix[i], "length"):
+                     lengths})
+            elif _is_scalar_data(dataset_hparams[i]["data_type"]):
+                # convert the list of strings into appropriate tensors here
+                data_type = dataset_hparams[i]["data_type"]
+                if data_type == "int":
+                    example_np = np.array(transposed_example, dtype=np.int32)
+                    example = torch.from_numpy(example_np). \
+                        to(device=self.device)
+                elif data_type == "float":
+                    example_np = np.array(transposed_example, dtype=np.float32)
+                    example = torch.from_numpy(example_np). \
+                        to(device=self.device)
+                else:
+                    raise ValueError(
+                        "Incorrect \'data_type\'. Currently \'int\' and "
+                        "\'float\' are supported. Received {}"
+                        .format(data_type))
+                batch.update({self._name_prefix[i]: example})
+            elif _is_record_data(dataset_hparams[i]["data_type"]):
+                for key, descriptor in \
+                        self._dataset_features[i]["feature_types"].items():  # type: ignore # pylint: disable=line-too-long
+                    values = [ex[key] for ex in transposed_example]
+                    if descriptor.shape is not None:
+                        # FixedLenFeature, do not pad.
+                        # NumPy functions work on PyTorch tensors too.
+                        if len(descriptor.shape) > 0 \
+                                and descriptor.shape[0] is None:
+                            values, _ = padded_batch(values)
+                        else:
+                            values = np.stack(values, axis=0)
+                        if (not torch.is_tensor(values) and
+                                descriptor.dtype not in [np.str_, np.bytes_]):
+                            values = torch.from_numpy(values)
+                    else:
+                        # VarLenFeature, just put everything in a Python list.
+                        pass
+                    batch.update({connect_name(self._name_prefix[i], key):
+                                  values})
+        return Batch(len(examples),
+                     **batch)
+
+    @staticmethod
+    def _get_name_prefix(dataset_hparams):
```

Comment:
Where is this method used? This should be a private method but I don't see it used anywhere within the class.